### PR TITLE
Add halfpageup and halfpagedown commands

### PR DESCRIPTION
**Pull request by tremby**
Like vim's ^U and ^D.

Closes ambientsound/pms#62

### CODE REVIEW

File: commands/cursor.go [hunk 1]
```diff
@@ -71,6 +71,12 @@ func (cmd *Cursor) Parse() error {
 	case "pgdn", "pagedn", "pagedown":
 		_, y := songlistWidget.Size()
 		cmd.relative = y
+	case "halfpageup":
+		_, y := songlistWidget.Size()
+		cmd.relative = -y / 2
+	case "halfpagedown":
```
Comment (ambientsound) [inline commands/cursor.go]: Suggest also adding `halfpgup`, `halfpagedn` and `halfpgdn`, to match the existing dictionary?

Comment (tremby) [inline commands/cursor.go]: Sure, will do so.

Comment (tremby) [inline commands/cursor.go]: Amended.

**Comment by tremby** [comment]
I would have liked to set these as default bindings ^U and ^D, but ^D is duplicate list by default. Any chance you want to move duplicate to another binding?

**Comment by ambientsound** [comment]
There are already bindings for full page up and down. Is half page up and down useful enough to warrant default bindings with the control keys? These are scarce and might be better suited for other purposes?

In any case, I'm open for concrete suggestions.

**Comment by tremby** [comment]
I use vim 8+ hours a day, and for me, a music player which says it's vim-like should feel like home. I use ^U and ^D a lot in vim (after pressing once you still have context you could see before, unlike ^F and ^B, and in vim they leave the cursor in place where possible, which is never possible with ^F and ^B -- that's on my mental list to fix in PMS once the scroll-while-keeping-cursor-in-place is implemented), and so I want to use them in PMS too!

If you think that's just me, that's OK: I can rebind just for myself.

What do you use duplicate list for? I've never used lists much.

**Comment by ambientsound** [comment]
You certainly have a point, and whether or not I think your usage of ^U and ^D is anecdotal is irrelevant since it should, as you say, feel at home.

Uncertain about binding ^K. How about ^R for "replicate", is this used anywhere in Vim?

**Comment by ambientsound** [comment]
^R is used for "redo" in Vim. Undo/redo might be implemented some time in the future. Maybe ^K is the better candidate. Any other suggestions?

**Comment by tremby** [comment]
Yeah I was going to say it might be nice to reserve that for future redo!

Hm, no other suggestions. All I can think of is things which would open another buffer in vim, like :new, :tabnew, etc. Could be something with a ^W prefix, like all the window-control functions in vim. `^W d` or `^W n` maybe? Grasping at straws.

**Comment by ambientsound** [comment]
The `^W` prefix sounds like a really good idea for many kinds of list manipulation. Let's remap `list duplicate` to `<C-w>d`?

**Comment by tremby** [comment]
OK, here it is. I threw in ^F and ^B as aliases for page up and page down while I was at it, since I think they belong.

That single ^W command seems lonely. I'd vote for moving (or duplicating) ^G to ^Wq (closes a window in vim) as a first step. But that might be a discussion for elsewhere.

**Pull request merged**